text/0000-throw-expr.md

```diff
@@ -0,0 +1,923 @@
+- Feature Name: `throw_expr`
+- Start Date: 2018-04-30
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Introduce diverging `throw expr` (or `fail expr` expressions), typed at `!`,
```

Comment:
"fail" doesn't seem like the right kind of word in terms of use of english. Consider: "throw an error", "raise an issue", "return an error", "report a problem". These all suggest an action taking place as a result of a failure. "fail" doesn't fit as well in this context, "failed to grab the lock", "fail this code section", "fail this function". It's more a statement that something went wrong than what action will be taken. e.g.
`// failed to open the file`
`raise "bad file name";`

My suggestion is to use "raise" as an alternative to "throw" instead of fail.

Reply:
Yes, [this was also pointed out](https://github.com/rust-lang/rfcs/pull/2426#issuecomment-386809498) by @ssokolow:
> That said, I'm not a fan of `fail` either because the intransitive use of the verb comes to mind first for me, which makes it feel odd.

In this regard, `raise` or `throw` are better. :+1:

Reply:
fun historical context: `panic!` was `fail!` before it was `panic!`.